File: tournaments/chartFix/chartFix.cs
int chartFix(int[] chart) {
    var n = chart.Length;
    var a = 1;
    var dp = new int[n];

    Array.Fill(dp, 1);

    for (var i = 1; n > i; ++i) {
        for (var j = i - 1; 0 <= j; --j)
            if (dp[j] + 1 > dp[i] && chart[j] < chart[i]) dp[i] = dp[j] + 1;
        a = Math.Max(a, dp[i]);
    }

    return n - a;
}

File: tournaments/chartFix/chartFix.cpp
int chartFix(std::vector<int> chart) {
  std::vector<int> toRemove;
  for (int i = 0; i < chart.size(); i++) {
    int cur = i;
    for (int j = 0; j < i; j++) {
      if (chart[j] < chart[i]) {
        cur = std::min(cur, toRemove[j] + i - j - 1);
      }
    }
    toRemove.push_back(cur);
  }
  int res = INT_MAX;
  for (int i = 0; i < toRemove.size(); i++) {
    res = std::min(res, toRemove[toRemove.size() - i - 1] + i);
  }
  return res;
}
